### DISCUSSION TITLE
receive commands by telnet

### DISCUSSION THREAD

**Discussion by lowepalma** [savonet/liquidsoap#3058] [User Support]
RELSOLVED !
I need help to understand how liquid soup can receive telnet commands to do functions.

Example but that it works correctly:

#Define a function that responds "hello" when called and checks the password
 def say_hello(password) =
            if password == "your_password" then
                "hello"
       else
         "Incorrect password"
      end
end

#Define the Telnet server and associate the "hello" command with the say_hello function
server = server.telnet(port = 1234)
server.register("hello", say_hello)

**Comment by vitoyucepi** [comment]
Hi @lowepalma,

You can find helpful information about registering the telnet commands  on page 152 of [the book](http://www.liquidsoap.info/book/book.pdf).

As for your example
```ruby
# Define a function that responds "hello" when called and checks the password
def say_hello(password)
  if password == "your_password" then
    "hello"
  else
    "Incorrect password"
  end
end

# Configure telnet server
settings.server.telnet.set(true)
# settings.server.telnet.revdns.set(false)
# settings.server.telnet.port.set(1234)
# settings.server.telnet.bind_addr.set("127.0.0.1")

# Associate the "hello" command with the say_hello function
server.register("hello", say_hello)
```

If you want to enable the telnet server, then you should use [settings](https://www.liquidsoap.info/doc-2.1.4/settings.html#support-for-communication-via-a-telnet-interface) instead of [`server.telnet`](https://www.liquidsoap.info/doc-2.1.4/reference.html#server.telnet).
The `server.telnet` function returns `unit`(`None` or `undefined`).
You should not assign it to the `server` variable because it's a global variable, and you will not be able to use it anymore.

According to the [function documentation](https://www.liquidsoap.info/doc-2.1.4/reference.html#server.register).
You have to supply
1. Name of the command as a string. In your case, it's `"hello"`.
1. Callback to function that consumes string and will return a string.

Additionally, you can provide
1. Namespace as a string. It will work as a prefix for your command.
1. Description as a string that will help the user to understand what the command will do.
1. Usage as a string. This will show the user how to use your command.

I suggest using [`harbor.http`](https://www.liquidsoap.info/doc-2.1.4/harbor_http.html) instead of telnet.
The information you're looking for can be found on page 156 of [the book](http://www.liquidsoap.info/book/book.pdf).
But it will [change significantly](https://www.liquidsoap.info/doc-dev/harbor_http.html)  in the upcoming release version *2.2*.

**Reply by lowepalma** [reply]
I would need a complete example, where I could see how it works, I am looking for liquidsoap to send an icecast url through a command, connect to that url, stop the list and then if I send another stop command, stop and continue the list. I can find an example somewhere.

**Reply by vitoyucepi** [reply]
I'm sorry, I'm not sure what you are looking for. Could you please clarify?
The workflow description appears to be a unique script.
I think it will require thoughtful research and analysis. Maybe you'll find some ready-to-use parts for your idea. But their recombination and compilation will be up to you.

**Reply by lowepalma** [reply]
Resolved !
